apps/prairielearn/src/pages/plrStudent/plrStudent.js

```diff
@@ -27,8 +27,11 @@ router.get('/', function (req, res, next) {
 
 
 // Function to get SEASONAL RESULTS
-function getSeasonalResults(callback) {
-  sqldb.query(sql.get_seasonal_results, [], function(err, result) {
+function getSeasonalResults(callback, res) {
```

Comment:
Instead of passing in the whole `res` object could you do something like this? Makes it easier to test rather than expecting that whole thing.

```js
var ERR = require('async-stacktrace');
// Routing Stuff
var express = require('express');
var router = express.Router();
// Query Stuff
var sqldb = require('@prairielearn/postgres');
var sql = sqldb.loadSqlEquiv(__filename);

// -------
// ROUTING
// -------
router.get('/', function (req, res, next) {
  var course_instance_id = res.locals.course_instance.id;
  getSeasonalResults(course_instance_id, function(err, results) {
    if (ERR(err, next)) return;
    res.locals.seasonalResults = results;
    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
  });
});

// ---------
// FUNCTIONS
// ---------
// TODO: Function to get USER INFO (user_id as parameter)


// TODO: Function to get LIVE RESULTS


// Function to get SEASONAL RESULTS
function getSeasonalResults(course_instance_id, callback) {
  sqldb.query(sql.get_seasonal_results, [course_instance_id], function(err, result) {
      if (ERR(err, callback)) return;
      callback(null, result.rows);
  });
}

// TODO: Function to get ALL-TIME RESULTS


module.exports = router;
```

Reply:
Or pass it in directly during the call:
```js
getSeasonalResults(res.locals.course_instance_id, function(err, results) 
```